WooCommerce/Classes/Analytics/WooAnalytics.swift

```diff
@@ -140,6 +140,7 @@ private extension WooAnalytics {
             return
         }
 
+        stopObservingNotifications()
```

Comment:
I believe the better course of action here might be to only call `startObservingNotifications()` once. I understand why we conditionally wrap the calls to loop into the NSNotificationCenter but in the end the `track()` method will ignore those events.

Reply:
So that means removing all of the `stopObservingNotifications()`, leave the one `startObservingNotifications()` in the `initialize()` method, and then add a `deinit` to call `stopObservingNotifications()`.

Reply:
You are right, I didn't notice that there was a check on the `track()` method. 
Since we target >= iOS 12.0 and we use `addObserver:selector:name:object:`, we don't need to unregister the observer inside the `deinit` method. The system will take care of it. It would have been different if we were using `addObserverForName:object:queue:usingBlock:`. 👍 
Can you take a second look at it?